SwiftCompilerSources/CMakeLists.txt

```diff
@@ -98,10 +98,17 @@ function(add_swift_compiler_modules_library name)
                         "DEPENDS"
                         ${ARGN})
 
+  # Prior to 5.9, we have to use the experimental flag for C++ interop.
+  if (CMAKE_Swift_COMPILER_VERSION VERSION_LESS 5.9)
```

Comment:
I don't think `CMAKE_Swift_COMPILER_VERSION` is valid unless we're building with hosttools. Last time I checked it was empty when bootstrapping is enabled since we aren't using CMake's built-in Swift support in that case. @eeckstein knows more about this.

Reply:
Perhaps this can be further refined to check something that indicates whether we are bootstrapping and specify `-cxx-interoperability-mode=default` when we are. Lmk if that sounds reasonable.

Reply:
That would make sense to me!

Reply:
I wouldn't bother with it. Bootstrapping is on life support and not even checked on the official CI: hosttools are de facto required at this point, with bootstrapping soon to be removed.

Reply:
Ok, given that assessment of bootstrapping, the fact that CI passed, and that this approach matches ASTGen I'm inclined to land it as-is. I think the failure case is just that we pass the deprecated flag when bootstrapping?